images: Split out the utility and packages to packagegroup and MACHINE_EXTRA_RRECOMMENDS

AGLExport opened the pull request:
Some packages forcefully install to the image using IMAGE_INSTALL. It causes loss of capability for image minimization and portability.

This patch changes to MACHINE_EXTRA_RRECOMMENDS and packagegroup from IMAGE_INSTALL.

NaotoYamaguchi-A committed:
images: Split out the utility and packages to packagegroup and MACHINE_EXTRA_RRECOMMENDS

Some packages forcefully install to the image using IMAGE_INSTALL.
It causes loss of capability for image minimization and portability.

This patch changes to MACHINE_EXTRA_RRECOMMENDS and packagegroup
from IMAGE_INSTALL.

Signed-off-by: Naoto Yamaguchi <naoto.yamaguchi@aisin.co.jp>

yhamamachi commented:
Hi @AGLExport,
I confirmed this PR breaks core-image-minimal build.
And, I made patch for this issue: https://github.com/rcar-community/meta-sparrow-hawk/commit/4ecaa849753b0e49b9bbd0cc68aeaaded9a0dc0c

If you have another solution, could you please push new patch?
If there is no response, I will release next version which contains these fixes tomorrow.

NaotoYamaguchi-A committed:
images: Fix for installation issue in core-image-minimal

The cd517081e4f815aae6ad00c1e0af9a298db8582f was a fixed kernel and some package
installation issue for many images.  On the other hand, core-image-minimal loses
the file under the /boot.
Those package installations are using MACHINE_EXTRA_RRECOMMENDS.  The package of
included MACHINE_EXTRA_RRECOMMENDS is installed by a dependency of
CORE_IMAGE_BASE_INSTALL.
This installation issue was caused by the core-image-minimal recipe that fourcelly
set IMAGE_INSTALL by itself.  As a result, CORE_IMAGE_BASE_INSTALL is removed
from IMAGE_INSTALL.

This patch adds CORE_IMAGE_BASE_INSTALL to core-image-minimal using the
existing .inc file.

Signed-off-by: Naoto Yamaguchi <naoto.yamaguchi@aisin.co.jp>

AGLExport commented:
I investigated it.  
The image.bbclass set CORE_IMAGE_BASE_INSTALL to IMAGE_INSTALL using weak.   The core-image-minimal.bb overwrites IMAGE_INSTALL by itself.  As a result, some board-specific packages were lost from core-image-minimal.  It is a specific issue for core-image-minimal.
When the image recipe has 'require core-image-minimal.bb', the same issue will reproduce.   Typical image uses 'require core-image-base.bb' such as core-image-weston and core-image-base.   This issue is not reproduced.

I will create a patch that includes:
IMAGE_INSTALL:append = " \
    ${CORE_IMAGE_BASE_INSTALL} \
 ”
in 
core-image-minimal.inc.

yhamamachi commented:
Thank you for your patch.
It looks good to me.
